tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/RegionDiscoveryRetryPolicyTests.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Http.Retry;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Region;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Unit.CoreTests;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.Identity.Test.Unit.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.ManagedIdentityTests
+{
+    [TestClass]
+    public class RegionDiscoveryRetryPolicyTests : RegionDiscoveryProviderTests
```

Comment:
Avoid inheritance. Just add more tests to the same file. You can have partial classes or regions if you feel there are too many tests in 1 file.